internal/rbd/types/manager.go

```diff
@@ -47,4 +49,8 @@ type Manager interface {
 	// DeleteVolumeGroup removes VolumeGroup from the backend storage and
 	// any details from the journal.
 	DeleteVolumeGroup(ctx context.Context, vg VolumeGroup) error
+
+	// GetMirrorSource returns the source of the mirror for the given volume or group.
+	GetMirrorSource(ctx context.Context, volumeID string,
+		rep *replication.ReplicationSource) ([]Volume, Mirror, error)
```

Comment:
it is not clear to me why `[]Volume` is returned. Should that not be either a single Volume or a VolumeGroup?

Reply:
would it make sense to have a `Mirror.GetSource()` function so that the API is easier to understand?

Reply:
Let's put everything related to processing volumes within functions in mirror type ?

Reply:
>would it make sense to have a Mirror.GetSource() function so that the API is easier to understand?

@nixpanic I put it with the Manager because that is the one who is generating the volume and group from an ID, moving this to Mirroring

>it is not clear to me why []Volume is returned. Should that not be either a single Volume or a VolumeGroup?

This is required because we need to work on all the volumes if its a group, i am trying to avoid type casting and doing the ListVolumes and generating the volumes again that contains the connections, let me know if you have any suggestions

> Let's put everything related to processing volumes within functions in mirror type ?

I mostly added all the mirroring specific things to mirror type but there are again generic one as well that the reason i left it in volume type as it makes more sense.

Reply:
:+1: 
We can check on volumes and refactor it later on.
Everything LGTM
Only concern now is ceph's API stability and go-ceph pr going in.

https://github.com/ceph/go-ceph/pull/1010#issuecomment-2223542910
https://github.com/ceph/ceph/pull/53793#discussion_r1670404359